01-ruby-fundamentals/keyword-arguments.md

```diff
@@ -129,16 +129,64 @@ coffee_price(:latte, :venti, 3, false)
 coffee_price(:drip, :tall, 0, true)
 ```
 
-There are a few problems with this:
-- The caller has to remember the order of the arguments
-- When reading the call, there's no way to tell which argument is which
-- The caller must provide values for all arguments, even for the "standard" way of doing things (i.e. no extra shots)
+Having so many parameters on one method introduces several issues:
+1. The caller must remember the order of the arguments
+2. When reading the call, there's no way to tell which argument is which
+3. The caller must provide values for all arguments, even for the "standard" way of doing things (i.e. no extra shots)
 
 All of these put the onus on the caller to do things right, which is just asking for trouble. **A well-written method is easy to use right and hard to use wrong** - by that standard, this is not good code.
 
+### Optional Arguments
+
+We're going to tackle #3 on the above list first by introducing ruby's support of **Optional Arguments**
+
+Optional arguments enable you to set default values when defining a function so that the caller doesn't need to pass them unless the value varies from the default. 
+
+```ruby
+# Define the method
+def coffee_price(type, size, extra_shots = 0, cold = false)
+  # ... calculate price, same code as before ...
+  return price
+end
+
+# Price for a venti latte with 3 extra shots
+coffee_price(:latte, :venti, 3)
+
+# Price for a tall iced drip
+coffee_price(:drip, :tall, 0, true)
+
+# Price for a grande cappuccino
+coffee_price(:cappuccino, :grande)
+```
```

Comment:
Will add an observations paragraph!
In reference to your 2nd paragraph: I considered calling that out but decided against it because I feared that without having introduced keyword arguments yet, students would struggle to follow the problem. Though convenient, it feels to me like more of a by-product of using the keyword and optional concepts together rather than the core reason either was introduced. 
That said, I think you worded it in about as comprehensible of a way as one could and I'm happy to add it there, just thought I'd give my concerns.

Reply:
I think it's probably fine to leave the second paragraph out - issues 1 and 2 are plenty of motivation for the next technique.